superx/information_extractor.py

```diff
@@ -22,133 +22,139 @@ class info_extractor:
         The data is then committed into the relevant table in the data base
         :param items: The child of the parsed xml tree containing all item information
         :param current_branch_id: id of current branch
         """
-        products_list = []
-        branch_price_list = []
-        item = 'Item'
-        bIsWeighted = 'bIsWeighted'
+        item_attr_name = self.current_super['item_attr_name']
+        bIsWeighted = self.current_super['is_weighted_attr_name']
 
-        # different supermarkets use different titles
-        if self.super_name == 'victory':
-            item = 'Product'
-            bIsWeighted = 'BisWeighted'
-
-        for item in items.findall(item):
-            code = int(item.find('ItemCode').text)
+        for item in items.findall(item_attr_name):
+            item_code = int(item.find('ItemCode').text)
             name = item.find('ItemName').text
-            quantity = item.find('Quantity').text
-            price = item.find('ItemPrice').text
+            # exclude unwanted names from DB
+            for exclude in self.exclude_names:
+                if exclude in name:
+                    continue
+
+            quantity = float(item.find('Quantity').text)
+            price = float(item.find('ItemPrice').text)
             update_date = self.standardize_date(item.find('PriceUpdateDate').text)
             is_weighted = False
-            if item.find(bIsWeighted).text == 1:
+            if int(item.find(bIsWeighted).text) == 1:
                 is_weighted = True
 
-            unit_of_measure = 'none'
-            if is_weighted:
-                unit_of_measure = self.convert_unit_name(item.find('UnitQty').text)
-
-            products_list.append(Product(id=code, name=name, quantity=quantity, is_weighted=is_weighted, unit_of_measure=unit_of_measure))
-            branch_price_list.append(BranchPrice(item_code=code, branch_id=current_branch_id, price=price, update_date=update_date))
-
-        # add all the product info to the db and commit
-        db.session.add_all(products_list)
-        db.session.commit()
-
-        # add all the branch price info to the db and commit
-        db.session.add_all(branch_price_list)
-        db.session.commit()
-
-    def get_branch_id(self, tree_children_list):
-        """
-        Gets branch id from a parsed xml tree
-        :param tree_children_list: list of the children of the root
-        :return: the branch id
-        """
-        for child in tree_children_list:
-            if child.tag == 'StoreId':
-                return child.text
+            unit_of_measure = self.convert_unit_name(item.find('UnitQty').text)
+            # if item already in db then continue to next item
+            if bool(Product.query.filter_by(id=item_code).first()):
+                # adding current_branch_price to the db and commit
+                current_branch_price = BranchPrice(item_code=item_code, price=price, update_date=update_date)
```

Comment:
In the code immediately after this, you add the product to the database and then the price.
Is that meant to be in an else?
In any case, If you erase this code and only add the product if it exists and then always add the price, that removes duplicate code.

Also, you probably need the BranchID in this table as well

Reply:
Same question as before. Where is the branchID?